Library/Homebrew/utils/github.rb

```diff
@@ -106,56 +107,60 @@ def api_credentials_error_message(response_headers)
     end
   end
 
-  def api_headers
-    {
-      "User-Agent" => HOMEBREW_USER_AGENT_RUBY,
-      "Accept"     => "application/vnd.github.v3+json"
-    }
-  end
-
   def open(url, &_block)
     # This is a no-op if the user is opting out of using the GitHub API.
     return if ENV["HOMEBREW_NO_GITHUB_API"]
 
-    require "net/https"
-
-    headers = api_headers
+    tmpfile = Tempfile.new("github_api_headers", HOMEBREW_TEMP)
```

Comment:
Maybe this is overkill, but the `Tempfile` documentation suggests to explicitly clean up the temporary file by doing something like this:

```ruby
tmpfile = Tempfile.new("github_api_headers", HOMEBREW_TEMP)
begin
  # Use `tmpfile`.
ensure
  tmpfile.close
  tmpfile.unlink
end
```

(It should be fine as-is, too, as it is being removed once the `Tempfile` instance is garbage collected.)